tracer/test/benchmarks/Benchmarks.Trace/OpenTelemetryAutoBenchmark.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.Collections.Specialized;
+using System.Diagnostics;
+using BenchmarkDotNet.Attributes;
+using Datadog.Trace;
+using Datadog.Trace.Activity.DuckTypes;
+using Datadog.Trace.Configuration;
+using OpenTelemetry;
+using OpenTelemetry.Trace;
+using OpenTelemetry.Resources;
+using ActivityListener = System.Diagnostics.ActivityListener;
+using ActivitySamplingResult = System.Diagnostics.ActivitySamplingResult;
+using ActivitySource = System.Diagnostics.ActivitySource;
+
+namespace Benchmarks.Trace;
+
+[MemoryDiagnoser]
+[BenchmarkAgent6]
+public class OpenTelemetryAutoBenchmark
+{
+    internal static ActivitySource ActivitySource { get; set; }
+    internal static Datadog.Trace.Tracer DatadogTracer { get; set; }
+    internal static OpenTelemetry.Trace.Tracer OpenTelemetryTracer { get; set; }
+    static OpenTelemetryAutoBenchmark()
+    {
+        ActivitySource = new ActivitySource("ActivityBenchmark");
+
+        var activityListener = new ActivityListener
+        {
+            ShouldListenTo = _ => true,
+            Sample = (ref ActivityCreationOptions<ActivityContext> options) => ActivitySamplingResult.AllData
+        };
+
+        ActivitySource.AddActivityListener(activityListener);
+
+        using var tracerProvider = Sdk.CreateTracerProviderBuilder()
+                                      .SetResourceBuilder(ResourceBuilder.CreateDefault().AddService("dd-trace-dotnet"))
+                                      .AddSource("ActivityBenchmark")
+                                      .AddConsoleExporter()
+                                      .Build();
+
+        OpenTelemetryTracer = tracerProvider.GetTracer("dd-trace-dotnet");
+        var environmentVars = new NameValueCollection();
+        environmentVars.Add("DD_TRACE_OTEL_ENABLED", "true");
+
+        var settings = new TracerSettings(new NameValueConfigurationSource(environmentVars)) { StartupDiagnosticLogEnabled = false };
+
+        Datadog.Trace.Tracer.UnsafeSetTracerInstance(new Datadog.Trace.Tracer(settings, new DummyAgentWriter(), null, null, null));
+        Datadog.Trace.ClrProfiler.Instrumentation.Initialize();
```

Comment:
Hmmm... I'm actually not sure how this will interact with the one other benchmark we have that calls `Datadog.Trace.ClrProfiler.Instrumentation.Initialize()`... (AspNetCoreBenchmarks)

By design, this method can only be called once, so if there's anything there which should be initialized differently based on settings, it won't be called in on or the other tests...

AFAIK, generally speaking, we avoid using auto instrumentation in the benchmarks, pretty much because of this 🤔 

Then again... if it works 🤷‍♂️  

(FWIW, our benchmarks seem to have been really variable recently, to the point that they're almost not useful 🙁 I initially thought it could be related to this PR, but I don't think that can be the case).

Reply:
Ahh I didn't know that we avoided auto instrumentation in benchmarks

Reply:
> Ahh I didn't know that we avoided auto instrumentation in benchmarks

I'm not really sure if we _explicitly_ do or not, it's just, it feels like it could cause instability... e.g. depending on the order benchmarks run, _sometimes_ you could be running benchmarks that are instrumented, and sometimes not. Although... now I think about it... we're not setting the profiler vars right so.... is this _actually_ instrumented anyway? 😅

Reply:
Hmm good point. When I copied it over the times I had in a separate project compared to these benchmarks seemed to be the same, so I just assumed it was working

Reply:
Commented out the benchmark attributes, I'll look into alternatives maybe just testing a few of the functions that do the `Activity`->`Span` mapping would be enough.